src/stories/Forms/InputWrapper/index.tsx

```diff
@@ -1,44 +1,147 @@
+    margin: calc(var(--spacing-8) * -1); // Negative margin to account for the padding, this is so the rest of the elements don't move
+    padding: var(--spacing-8);
+    border-radius: var(--border-radius);
+    background-color: var(--color-alertRedBg);`}
 `;
 
-interface InputWrapperProps {
-  className?: string;
-  children: JSX.Element | JSX.Element[];
+export interface InputWrapperProps {
   id: string;
-  alertLabel: string;
+  alert?: string;
+  children: JSX.Element | JSX.Element[];
+  className?: string;
+  hint?: string;
+  isRequired?: boolean;
 }
 
 /**
  * InputWrapper component
  */
 export const InputWrapper = ({
-  className,
-  children,
   id,
-  alertLabel,
+  alert,
+  children,
+  className,
+  hint,
+  isRequired,
 }: InputWrapperProps) => {
+  // Set aria id's. These are used for inputs and the helper texts.
+  const ariaDescribedById = hint ? `${id}-hint` : undefined;
+  const ariaErrorMessageId = alert ? `${id}-error` : undefined;
+
+  /*
+   * Define global child/input props.
+   * These are used to pass down to the children.
+   */
+  const globalInputProps = {
+    ariaDescribedBy: ariaDescribedById,
+    ariaErrorMessage: ariaErrorMessageId,
+    hasError: alert ? true : undefined,
+    id,
+    isRequired,
+  };
+
+  /**
+   * Pass globalInputProps to children.
+   * If the compontent is a fragment, we need to pass props to each child inside the fragment.
+   * However, if the child is not a valid react compontent, don't pass props at all.
+   */
+  const childrenWithGlobalInputProps = React.Children.map(children, (child) => {
+    if (child.type === React.Fragment) {
+      return React.Children.map(child.props.children, (child) => {
+        if (typeof child?.type === 'string') return child;
+        return React.cloneElement(child, globalInputProps);
+      });
+    }
+
+    if (typeof child?.type === 'string') return child;
+    return React.cloneElement(child, globalInputProps);
+  });
+
+  /*
+   * We apply special error styles for inputs such as checkboxes and radio buttons.
+   */
+  const hasBoxyErrorStyle = childrenWithGlobalInputProps?.some((child) => {
+    const componentsThatApply = [Checkbox, Radio];
+    return componentsThatApply.includes(child?.type);
+  });
+
   return (
-    <Wrapper className={className}>
-      {children}
-
-      {alertLabel && (
-        <InputError severity='error' id={`${id}-error`}>
-          {alertLabel}
-        </InputError>
-      )}
+    <Wrapper
+      className={className}
+      boxyErrorStyle={hasBoxyErrorStyle}
+      hasError={!!alert}
+    >
+      {childrenWithGlobalInputProps}
+
+      <div className='hint-wrapper'>
+        {
+          // Don't show the hint if there's an alert
+          hint && !alert && (
+            <span className='hint hint--help' id={ariaErrorMessageId}>
+              {hint}
+            </span>
+          )
+        }
+        {alert && (
+          <AlertText
+            className='hint hint--error'
```

Comment:
Can we do this without using className? Can it be a prop instead? It feels like we are breaking the pattern 🤔

Reply:
I mean I guess... But we will end up with a ton of styled functions. Will test it out tho. 👍